material-stepper/src/main/java/com/stepstone/stepper/type/TabsStepperType.java

```diff
@@ -57,11 +57,11 @@ public void onStepSelected(int newStepPosition) {
      * {@inheritDoc}
      */
     @Override
-    public void onNewAdapter(@NonNull AbstractStepAdapter<?> stepAdapter) {
+    public void onNewAdapter(@NonNull StepAdapter<?> stepAdapter) {
         List<Integer> titles = new ArrayList<>();
         final int stepCount = stepAdapter.getCount();
         for (int i = 0; i < stepCount; i++) {
-            final Step step = stepAdapter.getItem(i);
+            final Step step = (Step) stepAdapter.createStep(i);
```

Comment:
Due to the fact that StepAdapter no longer gets a <T extends Step> this is needed, however this feels kind of nasty. I'm wondering if there's a better way of handling this. @fabiozo @leonardo2204 any ideas?